pkg/cloud/api/diff.go

```diff
@@ -55,10 +67,11 @@ func (r *DiffResult) add(state DiffItemState, p Path, a, b reflect.Value) {
 		Path:  p,
 	}
 	if a.IsValid() {
-		di.A = a.Interface()
+		di.A = a.String()
```

Comment:
Why was the previous code incorrect? Are we seeing panics here?

From https://pkg.go.dev/reflect#Value.Interface

```
func (v [Value](https://pkg.go.dev/reflect#Value)) Interface() (i [any](https://pkg.go.dev/builtin#any))
Interface returns v's current value as an interface{}. It is equivalent to:

var i interface{} = (v's underlying value)
It panics if the Value was obtained by accessing unexported struct fields.
```

Reply:
We don't see panic here because type we use for rGraph does not have unexported fields. But don't we want to support that case?

Reply:
It's not very obvious this was your intent for the change.

1. You should not convert to String(). This violates the contract -- the type in the `di` struct is a `interface{}` and you are silently stringifying it in this layer.
2. If the check is to skip unexported fields, you need to put a comment here as it is not obvious that was the intent.
3. You need to add a test case that exercises the path.

Reply:
I added comment and if for unexported type to convert to string so in this case we will have some error reporting.